pmlc/rt/vulkan/vulkan_runtime.cc

```diff
@@ -54,6 +59,11 @@ class VulkanRuntime final : public pmlc::rt::Runtime {
   std::vector<std::shared_ptr<VulkanDevice>> devices;
 };
 
-pmlc::rt::RuntimeRegistration<VulkanRuntime> reg{"vulkan"};
+extern void registerSymbols();
+
+void registerRuntime() {
+  registerSymbols();
+  registerRuntime("vulkan", std::make_shared<VulkanRuntime>());
```

Comment:
We could just put a try/catch around this line I think.

Reply:
Actually, perhaps we should do it like:
```
try {
  registerRuntime("vulkan", std::make_shared<VulkanRuntime>());
  registerSymbols();
} catch (std::exception& ex) {
  IVLOG(2, "Failed to register 'vulkan' runtime: " << ex.what());
}
```

This way we only call `registerSymbols` if the runtime can be constructed.

Reply:
Yes could be, however I wanted to be consistent with the OpenCL runtime as is. If you think it is important I can move it to this place.